Makefile

```diff
@@ -35,11 +35,23 @@ clean:
 	rm -rf ./bin
 
 ci-preflight-checks:
+	$(MAKE) check-pipelinefiles
 	$(MAKE) check-dockerfiles
 	$(MAKE) check-language
 	$(MAKE) generate
 	$(MAKE) check-dirty
 
+pipeline-validator: hack/pipeline/*.go
+	docker run --rm \
+		-v $(CURDIR):/calico:rw \
+		-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
+		-w /calico/hack/pipeline \
+		$(CALICO_BUILD) go build -o /calico/bin/pipeline-validator .
```

Comment:
we could just make this `go run` instead of `go build` and it would save us a `make` target.